File: sdks/python/examples/basic.py
"""
Basic example using TAS Python SDK
"""
from tas_sdk import TASClient

# Initialize client
client = TASClient(
    api_key="your-api-key-here",
    base_url="https://tas.fly.dev"
)

# Example 1: Classify spam
print("Example 1: Spam detection")
result = client.classify(
    text="Earn money from home! Click here https://spam.com",
    lang="en"
)
print(f"  Is spam: {result['is_spam']}")
print(f"  Confidence: {result['confidence']:.3f}")
print(f"  Reason: {result['reason']}")
print()

# Example 2: Legitimate message
print("Example 2: Legitimate message")
result = client.classify(
    text="Hello, how are you? Want to grab coffee?",
    lang="en"
)
print(f"  Is spam: {result['is_spam']}")
print(f"  Confidence: {result['confidence']:.3f}")
print(f"  Reason: {result['reason']}")
print()

# Example 3: Russian spam
print("Example 3: Russian spam")
result = client.classify(
    text="Продам iPhone 12, цена 25000 руб. Срочно!",
    lang="ru"
)
print(f"  Is spam: {result['is_spam']}")
print(f"  Confidence: {result['confidence']:.3f}")
print(f"  Reason: {result['reason']}")
print()

# Example 4: Health check
print("Example 4: Health check")
health = client.health()
print(f"  Status: {health['status']}")
print(f"  Version: {health['version']}")
print(f"  LLM enabled: {health['llm_enabled']}")
print()

# Example 5: Version info
print("Example 5: Version info")
version = client.version()
print(f"  API Version: {version['api_version']}")
print(f"  Version: {version['version']}")

File: sdks/nodejs/examples/basic.js
/**
 * Basic example using TAS Node.js SDK
 */
const { TASClient } = require('../index');

async function main() {
    // Initialize client
    const client = new TASClient(
        'your-api-key-here',
        'https://tas.fly.dev'
    );

    // Example 1: Classify spam
    console.log('Example 1: Spam detection');
    let result = await client.classify(
        'Earn money from home! Click here https://spam.com',
        'en'
    );
    console.log(`  Is spam: ${result.is_spam}`);
    console.log(`  Confidence: ${result.confidence.toFixed(3)}`);
    console.log(`  Reason: ${result.reason}`);
    console.log();

    // Example 2: Legitimate message
    console.log('Example 2: Legitimate message');
    result = await client.classify(
        'Hello, how are you? Want to grab coffee?',
        'en'
    );
    console.log(`  Is spam: ${result.is_spam}`);
    console.log(`  Confidence: ${result.confidence.toFixed(3)}`);
    console.log(`  Reason: ${result.reason}`);
    console.log();

    // Example 3: Russian spam
    console.log('Example 3: Russian spam');
    result = await client.classify(
        'Продам iPhone 12, цена 25000 руб. Срочно!',
        'ru'
    );
    console.log(`  Is spam: ${result.is_spam}`);
    console.log(`  Confidence: ${result.confidence.toFixed(3)}`);
    console.log(`  Reason: ${result.reason}`);
    console.log();

    // Example 4: Health check
    console.log('Example 4: Health check');
    const health = await client.health();
    console.log(`  Status: ${health.status}`);
    console.log(`  Version: ${health.version}`);
    console.log(`  LLM enabled: ${health.llm_enabled}`);
    console.log();

    // Example 5: Version info
    console.log('Example 5: Version info');
    const version = await client.version();
    console.log(`  API Version: ${version.api_version}`);
    console.log(`  Version: ${version.version}`);
}

main().catch(console.error);

